app/plugin_key_value_store.go

```diff
@@ -97,6 +97,30 @@ func (a *App) GetPluginKey(pluginId string, key string) ([]byte, *model.AppError
 	return nil, nil
 }
 
+func (a *App) ModifyKey(pluginId string, key string, f func(value []byte) ([]byte, error)) *model.AppError {
+	kv, err := a.Srv.Store.Plugin().Get(pluginId, key)
+	if err != nil {
+		if err.StatusCode != http.StatusNotFound {
```

Comment:
Just a note, this is really bizarre in my opinion that we use an http status code to indicate that a `store` not found. @hanzei do you know if this is a common pattern? I thought the convention was that not found returns a nil value, and nil err.

Reply:
This is instead the one command and consistent pattern. I agree that it's bizarre, but this is how all code works right now.